### PR TITLE
homework_14

### PR DESCRIPTION
**Pull request by VadymDemchenko**

### CODE REVIEW

File: HW14/homework_14.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import json
+import time
+import re
+from datetime import datetime
+
+
+def name_time_decorator(func):
```
Comment (klymenok) [inline HW14/homework_14.py:7]: Was it the task to create a decorator?

File: HW14/homework_14.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    elif command == 'show_name':
+        name_info = input('Insert contact name to check info:\n')
+        print(contacts.get(name_info))
+    elif command == 'show_all':
+        print(contacts)
+    else:
+        print("\033[31m{}".format("Command unknown. Try one more time"))
+
+
+    class MyCustomException(Exception):
```
Comment (klymenok) [inline HW14/homework_14.py]: Was it the task to create a custom exception?

File: HW14/homework_14.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+            phone = input('Insert contact\'s phone in format \"+380xxxxxxxxx\" :\n')
+            ukr_phones = re.match(pattern, phone)
+            if ukr_phones:
+                contacts[name_add] = ukr_phones.group()
+                write(contacts, file_name)
+            else:
+                print(f"Invalid format of {phone} phone number. You can use:"
+                      "+380********* or "
+                      "380********* or "
+                      "80********* or "
```
Comment (klymenok) [inline HW14/homework_14.py]: This is a wrong format for the phone number. Check the task, please. And your pattern shouldn't support it.

**Review by klymenok — CHANGES_REQUESTED** [review]

File: HW14/homework_14.py (hunk 1 above)
Comment (VadymDemchenko) [reply HW14/homework_14.py:7]: Decorator is a part of contact book app. It make a log of each "read/write" function in app.

File: HW14/homework_14.py (hunk 2 above)
Comment (VadymDemchenko) [reply HW14/homework_14.py]: You're right. I'll delete it

File: HW14/homework_14.py (hunk 3 above)
Comment (VadymDemchenko) [reply HW14/homework_14.py]: Got it. Will be solved

**Review by klymenok — APPROVED** [review]
Looks good!

**Pull request merged**